lib/split/user.rb

```diff
@@ -55,32 +53,39 @@ def active_experiments
     end
 
     def alternative_key_for_experiment(experiment)
-      if experiment.version > 0
-        keys = user.keys
-
-        #default to current experiment key when one isn't found
-        user_experiment_key = experiment.key
-
-        #first version is not colon delimited 
-        if keys.include?(experiment.name)
-          user_experiment_key = experiment.name
-        else
-          experiment.version.times do |version_number|
-            key = "#{experiment.name}:#{version_number+1}"
-            if keys.include?(key)
-              user_experiment_key = key
-              break
-            end
-          end
-        end
+      user_experiment_key = first_field_from_all_versions(experiment)
+      #default to current experiment key when one isn't found
+      user_experiment_key || experiment.key
+    end
+
+    def all_fields_for_experiment_key(experiment_key)
+      user.keys - keys_without_experiment(user.keys, experiment_key)
+    end
 
-        user_experiment_key
+    def first_field_from_all_versions(experiment, exp_attribute = "")
```

Comment:
If `exp_attribute` is empty, it will look for the first appearance of the experiment key, like `exp805:2`. If `exp_attribute` has the value of `eligibility`, it will look for `exp805:2:eligibility`, with the first appearance from all the versions. We can reuse this method in `get_past_eligibility` from `cohoting_service`.